### DISCUSSION TITLE
I am forced to switch to Apollo Server from yoga, but Apollo expects different args to build server

### DISCUSSION THREAD

**Discussion by wakywayne** [hayes/pothos#651] [Q&A]
# The Problem:
### *I am forced to switch to Apollo Server as my yoga graphiql query interface is non-responsive despite deleting node_modules and .next folders. I am unable to type a word or move the insert cursor. I have had trouble shooted for a full day and now I have decided to just switch over to the apollo server*

--- 

### Apollo builds their server like this:
```
const server = new ApolloServer({
  typeDefs,
  resolvers,
});

const { url } = await startStandaloneServer(server, {
  listen: { port: 4000 },
});
```

### But my previous server with yoga was like this: 

```
const server = createServer({
    endpoint: '/api',
    schema: builder.toSchema(),
    context: async ({ req }) => {

        if (req.headers.user) {
            const user = await fetchUser(`${apiUrl}/api/auth/return/${req.headers.user}`);
            const parsedUser = JSON.parse(user);
            console.log({ parsedUser });

            return {
                currentUser: parsedUser
            }
        } else {
            null
        }
    }
})
```

### This is my schema builder incase it is relevant information

```
const builder = new SchemaBuilder<{
    Context: {
        currentUser: User;
    };

    AuthScopes: {
        public: boolean,
        user: boolean,
        admin: boolean,
    };

    Scalars: {
        Date: {
            Input: Date;
            Output: Date;
        };
        mongoId: {
            Input: ObjectId;
            Output: ObjectId;
        };
    };

    DefaultFieldNullability: true;

}>({
    plugins: [
        ScopeAuthPlugin,
        WithInputPlugin
    ],

    authScopes: async (context) => ({
        public: true,
        // eagerly evaluated scope
        user: context.currentUser ? context.currentUser.name == 'test' : false,
        // user: false,
        // evaluated when used 
        admin: false
        // scope loader with argument
    }),

    defaultFieldNullability: true,

    withInput: {
        // Go over this
        typeOptions: {
            // default options for Input object types created by this plugin
        },
        argOptions: {
            // set required: false to override default behavior
        },
    },
});
```

**Comment by Urigo** [comment]
@wakywayne I believe your issue was resolved?
Let me know if that's not the case
https://github.com/dotansimha/graphql-yoga/issues/1976

**Comment by wakywayne** [comment]
upgrading yoga to v3 resolved the issue yes thank you. HOWEVER, I still would like to know how to break up your pothos schemas into different files and import them. into the final schema, I can't find anything on this in the docs.

**Comment by hayes** [comment]
sorry I didn't get to this sooner, I've been traveling a bit, and haven't had as much time the last couple weeks.

Splitting up your schema into multiple files should be pretty straight forward.  Pothos isn't opinionated about how you do this.

The pattern I generally use is:

* Create a `builder.ts` file that crates your builder, and maybe adds some scalars, and an empty query and mutation type.
* create a directory of files for different types eg (user.ts, post.ts) that each define a graphql object type, along with any relevant queries/mutations (using `builder.queryField(s)` or `builder.mutationField(s)`).  
* create a `schema.ts` that imports each file above

```ts
import './types/user'
import './types/post'
import { builder } from './builder'
export const schema = builder.toSchema({})
```

There are some docs around this here:
https://pothos-graphql.dev/docs/guide/app-layout

**Reply by wakywayne** [reply]
I am sorry I appreciate you taking the time to answer my question. I am really new to this and I don't understand what the imports and exports would look like. Do you have a repository or code example that I can look at? I don't understand how the builder works exactly, so I think looking at the code broken up into it's parts would help me with that.

**Reply by hayes** [reply]
This example might help: https://github.com/hayes/pothos/tree/main/examples/complex-app/src/schema
